sls-versions/src/main/java/com/palantir/sls/versions/RegexMatcherParser.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.sls.versions;
+
+import static com.palantir.logsafe.Preconditions.checkNotNull;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+final class RegexMatcherParser {
```

Comment:
Should we move this to test sources?

Reply:
Done! Had to collapse the :benchmarks project into :sls-versions because it could no longer access this class once it was moved into :sls-versions' test sourceset, but I think this is fine because jmh stuff is separated into a nice jmh source set.